source/credentials_provider_process.c

```diff
@@ -0,0 +1,374 @@
+        }
+    }
+}
+
+static struct aws_string *s_get_command(struct aws_allocator *allocator, const struct aws_string *profile_to_use) {
+
+    bool success = false;
+    struct aws_string *command = NULL;
+    struct aws_profile_collection *config_profile = NULL;
+    struct aws_string *profile_name = NULL;
+    struct aws_profile *profile = NULL;
+
+    config_profile = s_load_profile(allocator);
+    if (!profile_to_use || !profile_to_use->len) {
+        profile_name = aws_get_profile_name(allocator, &s_default_profile_name_cursor);
+    } else {
+        profile_name = aws_string_new_from_string(allocator, profile_to_use);
+    }
+    if (config_profile && profile_name) {
+        profile = aws_profile_collection_get_profile(config_profile, profile_name);
+    }
+
+    if (!profile) {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "Failed to resolve config profile during process credentials provider initialization.");
+        goto on_finish;
+
+    } else {
+        s_check_or_get_with_profile_config(allocator, profile, &command, s_credentials_process);
+    }
+
+    if (!command || !command->len) {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "Failed to resolve credentials_process command during process credentials provider initialization.")
+        goto on_finish;
+    }
+
+    AWS_LOGF_DEBUG(
+        AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+        "Successfully loaded credentials_process command for process credentials provider.")
+    success = true;
+
+on_finish:
+    aws_string_destroy(profile_name);
+    aws_profile_collection_destroy(config_profile);
+    if (!success) {
+        aws_string_destroy(command);
+    }
+    return command;
+}
+
+static struct aws_credentials_provider_vtable s_aws_credentials_provider_process_vtable = {
+    .get_credentials = s_get_credentials_from_process,
+    .destroy = s_credentials_provider_process_destroy,
+};
+
+static struct aws_byte_cursor s_stderr_redirect_to_stdout = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL(" 2>&1");
+
+struct aws_credentials_provider *aws_credentials_provider_new_process(
+    struct aws_allocator *allocator,
+    const struct aws_credentials_provider_process_options *options) {
+
+    struct aws_credentials_provider *provider = NULL;
+    struct aws_credentials_provider_process_impl *impl = NULL;
+    struct aws_string *command = NULL;
+    struct aws_byte_buf command_with_stderr_redirected;
+    AWS_ZERO_STRUCT(command_with_stderr_redirected);
+
+    aws_mem_acquire_many(
+        allocator,
+        2,
+        &provider,
+        sizeof(struct aws_credentials_provider),
+        &impl,
+        sizeof(struct aws_credentials_provider_process_impl));
+
+    if (!provider) {
+        goto on_error;
+    }
+
+    AWS_ZERO_STRUCT(*provider);
+    AWS_ZERO_STRUCT(*impl);
+
+    command = s_get_command(allocator, options->profile_to_use);
```

Comment:
I would change command to a byte buf and have s_get_command write into it.  Then fold the stderr redirection into s_get_command by just doing an append to the impl's byte buf.